### ISSUE TITLE
[GCP] 0.7 - basic auth 502's backend is unhealthy

### ISSUE THREAD

**Issue by jlewi** [kubeflow/manifests#593]
Using basic auth on GCP with
kubeflow/manifests - v0.7.0-rc.2-16-gd6dad08
kubeflow/kubeflow - v0.4.0-rc.1-802-gd66f8b97

Accessing the central dasbhoard results in 502s.  The GCP backend reports 0 healthy backends.
It looks like the health check is set to "/".

I suspect this is related to kubeflow/manifests#513

**Comment by jlewi** [comment]
Pod is using

```
  serviceAccount: kf-admin
  serviceAccountName: kf-admin
```

Not seeing any rolebindings for it in the istio-namespace. Lets try creating one

```
kubectl -n istio-system create rolebinding kf-admin --serviceaccount=istio-system:kf-admin --clusterrole=kubeflow-admin
```

So that fixed the issue but now the code is stuck here unable to get a backend
https://github.com/kubeflow/manifests/blob/127dacce625e5dfc7630302ea82cf8ad56f9b6dd/gcp/basic-auth-ingress/base/config-map.yaml#L33

```
+ [[ -z '' ]]
++ kubectl --namespace=istio-system get ingress envoy-ingress -o 'jsonpath={.metadata.annotations.ingress\.kubernetes\.io/backends}'
+ BACKENDS='{"k8s-be-30893--4295f5d0f52d5379":"HEALTHY","k8s-be-31798--4295f5d0f52d5379":"UNHEALTHY"}'
+ echo 'fetching backends info with envoy-ingress: {"k8s-be-30893--4295f5d0f52d5379":"HEALTHY","k8s-be-31798--4295f5d0f52d5379":"UNHEALTHY"}'
fetching backends info with envoy-ingress: {"k8s-be-30893--4295f5d0f52d5379":"HEALTHY","k8s-be-31798--4295f5d0f52d5379":"UNHEALTHY"}
++ grep -o 'k8s-be---[0-9a-z]\+'
++ echo '{"k8s-be-30893--4295f5d0f52d5379":"HEALTHY","k8s-be-31798--4295f5d0f52d5379":"UNHEALTHY"}'
+ BACKEND_NAME=
+ echo 'backend name is '
+ sleep 2
backend name is 
```

**Comment by jlewi** [comment]
The code is 

```
 BACKENDS=$(kubectl --namespace=${NAMESPACE} get ingress ${INGRESS_NAME} -o jsonpath='{.metadata.annotations.ingress\.kubernetes\.io/backends}')
echo "fetching backends info with ${INGRESS_NAME}: ${BACKENDS}"
BACKEND_NAME=$(echo $BACKENDS | grep -o "k8s-be-${NODE_PORT}--[0-9a-z]\+")
```

So it looks like node port is empty.

**Comment by jlewi** [comment]
Looks like the code is slightly different from what we do for IAP.
https://github.com/kubeflow/manifests/blob/127dacce625e5dfc7630302ea82cf8ad56f9b6dd/gcp/iap-ingress/base/config-map.yaml#L81

It looks like we didn't get the node port because of the RBAC permission

```
++ kubectl --namespace=istio-system get svc ambassador -o 'jsonpath={.spec.ports[0].nodePort}'
Error from server (Forbidden): services "ambassador" is forbidden: User "system:serviceaccount:istio-system:kf-admin" cannot get resource "services" in API group "" in the namespace "istio-system"
+ NODE_PORT=
+ echo node port is
node port is
+ [[ -z '' ]]
```
I deleted the pod to force a restart and now the script worked and the backend was updated

The health check path ended up getting set to "/whoami". Look like that's what we set here
https://github.com/kubeflow/manifests/blob/127dacce625e5dfc7630302ea82cf8ad56f9b6dd/gcp/basic-auth-ingress/base/stateful-set.yaml#L27

After that I was able to login successfully.

So it looks like we have the following issue

1. The RBAC roles for kf-admin in istio-system namespace aren't set correctly
    * We should compare to what we do for IAP
1. backend-updater.sh should not keep going if the NODEPORT is empty
1. We need an endpoint test for basic_auth
